at_client/src/main/java/org/atsign/client/cli/Register.java

```diff
@@ -1,102 +1,167 @@
+                apiKey = configReader.getProperty("API_KEY");
+            }
         }
 
         if (rootDomain == null || rootPort == null || registrarUrl == null || apiKey == null) {
-            System.err.println("Please make sure to set all relevant configuration in src/main/resources/config.yaml");
+            System.err.println(
+                    "Please make sure to set all relevant configuration in src/main/resources/config.yaml");
             System.exit(1);
         }
 
-        Scanner scanner = new Scanner(System.in);
-        RegisterUtil registerUtil = new RegisterUtil();
+        if (!isRegistrarV3) {
+            Scanner scanner = new Scanner(System.in);
+            System.out.println("Getting free atsign");
+            atsign = new AtSign(registerUtil.getFreeAtsign(registrarUrl, apiKey));
+            System.out.println("Got atsign: " + atsign);
 
-        System.out.println("Getting free atsign");
-        AtSign atsign = new AtSign(registerUtil.getFreeAtsign(registrarUrl, apiKey));
-        System.out.println("Got atsign: " + atsign);
-
-        System.out.println("Sending one-time-password to :" + email);
-        if (registerUtil.registerAtsign(email, atsign, registrarUrl, apiKey)) {
-            System.out.println("Enter OTP received on: " + email);
-
-            otp = scanner.nextLine();
-            System.out.println("Validating one-time-password");
-            validationResponse = registerUtil.validateOtp(email, atsign, otp, registrarUrl, apiKey, false);
-            // if validationResponse is retry, the OTP entered is incorrect. Ask user to
-            // re-enter correct OTP
-            if ("retry".equals(validationResponse)) {
-                while ("retry".equals(validationResponse)) {
-                    System.out.println("Incorrect OTP entered. Re-enter the OTP: ");
-                    otp = scanner.nextLine();
-                    validationResponse = registerUtil.validateOtp(email, atsign, otp, registrarUrl, apiKey, false);
+            System.out.println("Sending one-time-password to :" + email);
+            if (registerUtil.registerAtsign(email, atsign, registrarUrl, apiKey)) {
+                System.out.println("Enter OTP received on: " + email);
+
+                otp = scanner.nextLine();
+                System.out.println("Validating one-time-password");
+                validationResponse = registerUtil.validateOtp(email, atsign, otp, registrarUrl, apiKey, false);
+                // if validationResponse is retry, the OTP entered is incorrect. Ask user to
+                // re-enter correct OTP
+                if ("retry".equals(validationResponse)) {
+                    while ("retry".equals(validationResponse)) {
+                        System.out.println("Incorrect OTP entered. Re-enter the OTP: ");
+                        otp = scanner.nextLine();
+                        validationResponse = registerUtil.validateOtp(email, atsign, otp, registrarUrl, apiKey, false);
```

Comment:
should there me max retry count?..otherwise we will end up calling registerUtil.validateOtp forever

Reply:
the latest commit has max retries of 3